contracts/modules/grouping/GroupingModule.sol

```diff
@@ -0,0 +1,220 @@
+    constructor(
+        address accessController,
+        address ipAssetRegistry,
+        address licenseRegistry,
+        address royaltyModule,
+        address groupNFT
+    ) AccessControlled(accessController, ipAssetRegistry) {
+        if (royaltyModule == address(0)) revert Errors.GroupingModule__ZeroRoyaltyModule();
+        if (licenseRegistry == address(0)) revert Errors.GroupingModule__ZeroLicenseRegistry();
+        if (groupNFT == address(0)) revert Errors.GroupingModule__ZeroGroupNFT();
+        if (ipAssetRegistry == address(0)) revert Errors.GroupingModule__ZeroIpAssetRegistry();
+
+        ROYALTY_MODULE = IRoyaltyModule(royaltyModule);
+        GROUP_IP_ASSET_REGISTRY = IGroupIPAssetRegistry(ipAssetRegistry);
+        LICENSE_REGISTRY = ILicenseRegistry(licenseRegistry);
+
+        if (!groupNFT.supportsInterface(type(IGroupNFT).interfaceId)) {
+            revert Errors.GroupingModule__InvalidGroupNFT(groupNFT);
+        }
+        GROUP_NFT = IGroupNFT(groupNFT);
+        _disableInitializers();
+    }
+
+    /// @notice initializer for this implementation contract
+    /// @param accessManager The address of the protocol admin roles contract
+    function initialize(address accessManager) public initializer {
+        if (accessManager == address(0)) {
+            revert Errors.GroupingModule__ZeroAccessManager();
+        }
+        __ReentrancyGuard_init();
+        __UUPSUpgradeable_init();
+        __ProtocolPausable_init(accessManager);
+    }
+
+    /// @notice Registers a Group IPA.
+    /// @param groupPool The address of the group pool.
+    /// @return groupId The address of the newly registered Group IPA.
+    function registerGroup(address groupPool) external whenNotPaused returns (address groupId) {
+        // mint Group NFT
+        uint256 groupNftId = GROUP_NFT.mintGroupNft(msg.sender, address(this));
+        // register Group NFT
+        groupId = GROUP_IP_ASSET_REGISTRY.registerGroup(address(GROUP_NFT), groupNftId, groupPool);
+        // initialize royalty vault
+        // transfer all royalty tokens to the  group pool
+        // transfer Group NFT to msg.sender
+        GROUP_NFT.safeTransferFrom(address(this), msg.sender, groupNftId);
+        emit IPGroupRegistered(groupId, groupPool);
+    }
+
+    /// @notice Whitelists a group reward pool.
+    /// @param rewardPool The address of the group reward pool.
+    function whitelistGroupRewardPool(address rewardPool) external restricted {
+        GROUP_IP_ASSET_REGISTRY.whitelistGroupRewardPool(rewardPool);
+    }
+
+    /// @notice Adds IP to group.
+    /// the function must be called by the Group IP owner or an authorized operator.
+    /// @param groupIpId The address of the group IP.
+    /// @param ipIds The IP IDs.
+    function addIp(address groupIpId, address[] calldata ipIds) external whenNotPaused verifyPermission(groupIpId) {
+        if (LICENSE_REGISTRY.hasDerivativeIps(groupIpId)) {
```

Comment:
same as https://github.com/storyprotocol/protocol-core-v1/pull/192/files#r1729601324

Reply:
Add restriction that lock the group when it has minted any license token.